.gitlab-ci.yml

```diff
@@ -75,6 +74,9 @@ publish_public_main:
     - if: '$CI_COMMIT_BRANCH == "main"'
       when: on_success
     - when: never
+  needs:
+    - build_image_arm64
```

Comment:
Now that you removed the `intermediate` stage, this is not required anymore. (in all 3 release jobs)

Reply:
for my knowledge, if we don't provide a list of `needs` it means all the jobs from the previous "stage" need to succeed, correct?

Reply:
Yes, which is what we want in this case, everything in `image` needs to be successful to run `publish`